analyzer/src/main/kotlin/managers/utils/NuGetInspector.kt

```diff
@@ -0,0 +1,324 @@
+            party.name?.let { append(it) }
+            party.email?.let {
+                append(if (party.name != null && it.isNotBlank()) " <$it>" else it)
+            }
+        }.takeIf { it.isNotBlank() }
+    }
+
+private fun NuGetInspector.PackageData.toPackageReference() =
+    PackageReference(
+        id = Identifier(type = TYPE, namespace = "", name = name, version = version),
+        dependencies = dependencies.mapTo(sortedSetOf()) {
+            PackageReference(
+                id = Identifier(type = TYPE, namespace = "", name = it.name, version = it.version),
+            )
+        }
+    )
+
+internal fun NuGetInspector.Result.toOrtProject(
+    managerName: String,
+    analysisRoot: File,
+    definitionFile: File,
+): Project {
+    val spec = resolveLocalSpec(definitionFile)?.let { NuGetSupport.XML_MAPPER.readValue<PackageSpec>(it) }
+    val id = Identifier(
+        type = managerName,
+        namespace = "",
+        name = spec?.metadata?.id ?: definitionFile.relativeTo(analysisRoot).invariantSeparatorsPath,
+        version = spec?.metadata?.version.orEmpty()
+    )
+    val workingDir = definitionFile.parentFile
+
+    val nestedPackages: MutableList<NuGetInspector.PackageData> = mutableListOf()
+
+    packages.forEach {
+        it.packages.forEach { pkg -> nestedPackages.add(pkg) }
+    }
+
+    val packageReferences = nestedPackages.mapTo(sortedSetOf()) {
+        it.toPackageReference()
+    }
+
+    val scopes = sortedSetOf(Scope(headers.first().projectFramework, packageReferences))
+    return Project(
+        id = id,
+        definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
+        vcs = VcsInfo.EMPTY,
+        authors = parseAuthors(spec),
+        vcsProcessed = PackageManager.processProjectVcs(workingDir),
+        declaredLicenses = parseLicenses(spec),
+        homepageUrl = "",
+        scopeDependencies = scopes
+    )
+}
+
+internal fun Set<NuGetInspector.PackageData>.toOrtPackages(): Set<Package> =
+    groupBy { "${it.name}:${it.version}" }.mapTo(mutableSetOf()) { (_, packages) ->
+        val pkg = packages.first()
+
+        fun NuGetInspector.PackageData.getHash(): Hash = Hash.create(
+            sha512?.lowercase() ?: sha256?.lowercase() ?: sha1?.lowercase() ?: md5?.lowercase() ?: ""
+        )
+
+        val id = Identifier(type = TYPE, namespace = pkg.namespace.orEmpty(), name = pkg.name, version = pkg.version)
+        val declaredLicenses: SortedSet<String> = sortedSetOf()
+        val pkgDeclaredLicense = pkg.declaredLicense.orEmpty()
+
+        if (pkgDeclaredLicense.isNotBlank()) {
+            val licenseData = yamlMapper.readValue<Map<String, String>>(pkgDeclaredLicense)
+            val type = licenseData["LicenseType"].orEmpty()
+            licenseData["LicenseExpression"]?.also { if ( type == "Expression" ) declaredLicenses += it }
+            licenseData["LicenseUrl"]?.also { if ( declaredLicenses.isEmpty() ) declaredLicenses += it }
```

Comment:
If the idea is to only use "LicenseUrl" as a fallback, you may use something like

    listOfNotNull(
        licenseData["LicenseExpression"],
        licenseData["LicenseUrl"]
    ).firstOrNull()?.also { declaredLicenses += it }

to make that more clear.

Reply:
@sschuberth thanks! how can we take care of the `type == "Expression"` condition in this as well ?

Reply:
Use `licenseData["LicenseExpression"]?.takeIf { type == "Expression" }`.

Reply:
Done!